django/db/models/functions/window.py

```diff
@@ -0,0 +1,119 @@
+from django.db.models import FloatField, Func, IntegerField
+
+__all__ = [
+    'CumeDist', 'DenseRank', 'FirstValue', 'Lag', 'LastValue', 'Lead',
+    'NthValue', 'Ntile', 'PercentRank', 'Rank', 'RowNumber',
+]
+
+
+class WindowFunction(Func):
+    window_compatible = True
+
+
+class CumeDist(WindowFunction):
+    function = 'CUME_DIST'
+    name = 'CumeDist'
+
+    def __init__(self, *expressions, **extra):
+        super().__init__(*expressions, output_field=FloatField())
```

Comment:
See https://github.com/django/django/pull/7611/files#r131567089

We should make this easier - but in a separate patch. There's already a def output_field and def _output_field.. we can fix it all up at once.